test/test_autograd.py

```diff
@@ -7066,6 +7066,20 @@ def test_min_max_median_backprops_to_all_values(self, device):
                 self.assertEqual(x.grad.sum(), 1.)
                 self.assertEqual((x.grad == 1 / 3).sum(), 3)
 
+    def test_scatter_reduce_prod_gradgrad_error(self, device):
+        # test that double backward raises an error
+        input = torch.tensor([1.], device=device, dtype=torch.float64, requires_grad=True)
+        src = torch.tensor([0., 0.], device=device, dtype=torch.float64, requires_grad=True)
+        idx = torch.tensor([0, 0], device=device, dtype=torch.long)
+
+        def fn(input, src):
+            return input.scatter_reduce(0, idx, src, 'prod')
+
+        # check that gradcheck doesn't raise an error
```

Comment:
nit: we actually want to make sure that gradcheck passes here are it is the only place where we check it for this case!